p4-16/spec/P4-16-spec.adoc

```diff
@@ -5498,6 +5498,12 @@ types (e.g. ``struct``s) are copied recursively, and all components
 of ``header``s are copied, including "validity" bits. Assignment is
 not defined for `extern` values.
 
+An assignment may also be written with a binary operator immediately
+before the `=` sign.  This performs the binary operator on the old
+value of the left sub-expression and the right sub-expression and assigns
+the result to the l-value.  Thus an assignment like `A += B` is equivalent
+to `A = A + B`, except that `A` is only evaluated once.
```

Comment:
I would explicitly mention that any side effects that may be in A are performed just once (since P4 does not have any concept of references, it may not be entirely clear what an lvalue is evaluated to).